contracts/MagicatsHandlerUpgradeable.sol

```diff
@@ -176,6 +183,7 @@ contract MagicatsHandlerUpgradeable is
         for (uint256 i = 0; i < IDsToStake.length; i = _uncheckedInc(i)) {
             magicatIdToStakedPid[IDsToStake[i]] = poolID;
         }
+        lastAllocationTimestamp = block.timestamp;
```

Comment:
this internal function is called anytime someone deposits or withdraws. I think you want to move this to the external keeper-only function.

Reply:
also should we change the signature of that function to something like:

```
function updateStakedMagicats(
        uint256 poolID,
        uint256[] memory IDsToStake,
        uint256[] memory IDsToUnstake,
        bool allocationCompleted
    )
```

that way you set the timestamp only on the last TX of a batch?

Reply:
Yes that could be useful. With 5 tx lets say the 4th fails then the timestamp will not be set and it would run again the next time. This would add to the gas cost but make the system better able to recover from errors which may be more important.